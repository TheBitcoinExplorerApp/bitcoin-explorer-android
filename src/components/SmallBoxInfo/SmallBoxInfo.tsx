/* eslint-disable import/no-extraneous-dependencies */
import { View, Text, StyleSheet } from "react-native";

export type SmallBox = {
  title: string;
  value: string;
};

type SmallBoxInfoProps = {
  boxesInfos: SmallBox[];
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-around",
  },
  boxContainer: {
    textAlign: "center",
    alignItems: "center",
    backgroundColor: "#1d2133",
    width: 100,
    paddingVertical: 16,
    paddingHorizontal: 8,
    borderRadius: 7,
  },
  text: {
    color: "#FFF",
    fontSize: 14,
  },
});

export default function SmallBoxInfo(props: SmallBoxInfoProps) {
  const { boxesInfos } = props;

  return (
    <View style={styles.container}>
      {boxesInfos.map((infos, idx) => (
        // eslint-disable-next-line react/no-array-index-key
        <View style={styles.boxContainer} key={idx}>
          <Text style={styles.text}>{infos.title} sat/vB</Text>
          {/* <Text style={styles.text}>R$ {infos.value}</Text>
            #TODO: take price value from another endpoint to calculate the value in BRL
          */}
        </View>
      ))}
    </View>
  );
}
