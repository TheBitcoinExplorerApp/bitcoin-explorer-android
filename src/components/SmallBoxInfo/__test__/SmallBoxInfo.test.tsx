/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import SmallBoxInfo from "../SmallBoxInfo";

describe("SmallBoxInfo", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <SmallBoxInfo
          boxesInfos={[
            { title: "1", value: "0.000001" },
            { title: "2", value: "0.000002" },
            { title: "3", value: "0.000003" },
          ]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
