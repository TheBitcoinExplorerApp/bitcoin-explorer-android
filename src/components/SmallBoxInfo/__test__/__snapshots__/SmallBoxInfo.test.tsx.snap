// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallBoxInfo renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1d2133",
        "borderRadius": 7,
        "paddingHorizontal": 8,
        "paddingVertical": 16,
        "textAlign": "center",
        "width": 100,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 14,
        }
      }
    >
      1
       sat/vB
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1d2133",
        "borderRadius": 7,
        "paddingHorizontal": 8,
        "paddingVertical": 16,
        "textAlign": "center",
        "width": 100,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 14,
        }
      }
    >
      2
       sat/vB
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1d2133",
        "borderRadius": 7,
        "paddingHorizontal": 8,
        "paddingVertical": 16,
        "textAlign": "center",
        "width": 100,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 14,
        }
      }
    >
      3
       sat/vB
    </Text>
  </View>
</View>
`;
